pkg/ineffassign/ineffassign.go

```diff
@@ -279,7 +285,9 @@ func (bld *builder) Visit(n ast.Node) ast.Visitor {
 		}
 		if ok && n.Op == token.AND {
 			if v, ok := bld.vars[id.Obj]; ok {
-				v.escapes = true
+				if !checkEscapingErrors && id.Obj.Name != "err" {
```

Comment:
This should be
```suggestion
				if !(checkEscapingErrors && id.Obj.Name == "err") {
```

Reply:
And there are two other places where `escapes` is set to `true`.  Define and use a method:
```go
func (bld *builder) escape(id *ast.Ident) {
	if checkEscapingErrors && id.Obj.Name == "err" {
		return
	}
	if v, ok := bld.vars[id.Obj]; ok {
		v.escapes = true
	}
}
```